Assets/Countly/Plugins/CountlySDK/Services/UserProfile.cs

```diff
@@ -0,0 +1,682 @@
+    /// <param name="key">string with property name for array property</param>
+    /// <param name="value">string with value to add to array</param>
+    public void PushUnique(string key, string value)
+    {
+        Log.Info($"[UserProfile] Calling PushUnique with key: [{key}] and value: [{value}]");
+        PushUniqueInternal(key, value);
+    }
+
+    /// <summary>
+    /// Send provided values to server.
+    /// </summary>
+    public void Save()
+    {
+        Log.Info("[UserProfile] Calling Save");
+        SaveInternal();
+    }
+
+    /// <summary>
+    /// Sets user data values.
+    /// </summary>
+    /// <param name="userData">Dictionary with user data</param>
+    public void SetData(Dictionary<string, object> userData)
+    {
+        Log.Info("[UserProfile] Calling SetData for " + userData.Count + " amount of values");
+        SetDataInternal(userData);
+    }
+
+    /// <summary>
+    /// Set value only if property does not exist yet.
+    /// </summary>
+    /// <param name="key">string with property name to set</param>
+    /// <param name="value">value string value to set</param>
+    public void SetOnce(string key, string value)
+    {
+        Log.Info($"[UserProfile] Calling SetOnce with key: [{key}] and value: [{value}]");
+        SetOnceInternal(key, value);
+    }
+
+    /// <summary>
+    /// Provide a Dictionary of user properties to set.
+    /// Those can be either custom user properties or predefined user properties.
+    /// </summary>
+    /// <param name="data">Dictionary with data to set</param>
+    public void SetProperties(Dictionary<string, object> data)
+    {
+        Log.Info("[UserProfile] Calling SetProperties for " + data.Count + " amount of values");
+        SetPropertiesInternal(data);
+    }
+
+    /// <summary>
+    /// Set a single user property. It can be either a custom one or one of the predefined ones.
+    /// </summary>
+    /// <param name="key">string with key for the user property</param>
+    /// <param name="value">value for the user property to be set. The value should be the allowed data type.</param>
+    public void SetProperty(string key, object value)
+    {
+        Log.Info($"[UserProfile] Calling SetProperty with key: [{key}] and value: [{value}]");
+        SetPropertyInternal(key, value);
+    }
+    #endregion
+    #region Internal Calls
+    private void IncrementInternal(string key, double value)
+    {
+        if (!cly.IsSDKInitialized) {
+            Log.Warning("[UserProfile][IncrementInternal], Countly.Instance.Init() must be called before IncrementInternal");
+            return;
+        }
+        
+        if (!ValidateConsentAndKey(key, "IncrementInternal")) {
+            return;
+        }
+
+        Log.Info($"[UserProfile][IncrementInternal] key: [{key}] and value: [{value}]");
+        ModifyCustomData(key, value, "$inc");
+    }
+
+    private void MaxInternal(string key, double value)
+    {
+        if (!cly.IsSDKInitialized) {

```

Comment:
could not we move this is init check to the validation function, it looks like all the other methods are also checking this

Reply:
yea makes sense, let me do it